### DISCUSSION TITLE
When rabbitmq-server is started, there is a low probability that a udp socket listens to ports of all network interfaces. The rabbitmq version is 3.11.9, and the erlang version is 25.2.2.

### DISCUSSION THREAD

**Discussion by tkgup** [rabbitmq/rabbitmq-server#10467] [Other] [answered]
### Describe the bug

After the problem occurs, restart the server to solve the problem. This may be a probability event.
![bad_socket](https://github.com/rabbitmq/rabbitmq-server/assets/61499462/10db503e-f45a-42fa-b56d-52deee180bc1)

### Reproduction steps

1.start rabbitmq server
2.execute shell command: netstat -anpu | grep ${rabbitmq_pid}
3.There is a possibility that one all-zero UDP socket is monitored.

### Expected behavior

The network interface is stable and controllable, for example, through configuration.

**Comment by michaelklishin** [comment]
RabbitMQ nodes can open client sockets, some features very explicitly depends on its ability to do that.

OAuth 2 support, Shovel, Federation are some examples when the client sockets are TCP ones.

Nodes also periodically connect to [epmd](https://rabbitmq.com/networking.html#epmd) to find out if they are still registered with it. I could not find any evidence of it using UDP, the docs [do not make that clear](https://www.erlang.org/doc/apps/erts/erl_dist_protocol.html) but TCP is mentioned 9 times and UDP is not mentioned.

RabbitMQ never "calls home", so I don't know what may be opening those client sockets. You haven't shared any details about your node other than the version.

**Comment by lukebakken** [comment, marked as answer]
@tkgup you have not provided enough information for us to assist you:
* What is the operating system and version you're using?
* What version of Erlang?
* What version of RabbitMQ?
* How are you installing Erlang and RabbitMQ?
* How are you running RabbitMQ? Via `systemd`, k8s, something else?

**Reply by michaelklishin** [reply]
What plugins are enabled

**Reply by tkgup** [reply]
operating system： x86_64 and aarch64 have a probability of occurring on EulerOS system.
Erlang version: 25.2.2
RabbitMQ version: 3.11.9
install: Make rpm to install
running way: Not in the container, directly on the host, running with the default shell: rabbitmq-server
base info:
![image](https://github.com/rabbitmq/rabbitmq-server/assets/61499462/2e722242-4273-4223-806b-1edf9f88c2f3)

plugin info：
![image](https://github.com/rabbitmq/rabbitmq-server/assets/61499462/522fa0a1-1774-4a46-ac59-64ea4774e94a)

**Reply by tkgup** [reply]
according to disscusion ( https://github.com/rabbitmq/rabbitmq-server/issues/1718 ):
execute commond:
erl -sname tkg-test -setcookie rabbitmq_server_cookie -remsh rabbit@rabbitmqNode0
PortInfos = lists:map(fun erlang:port_info/1, erlang:ports()).
rp(PortInfos).
rp(erlang:process_info(erlang:list_to_pid("<0.800.0>"))).

![image](https://github.com/rabbitmq/rabbitmq-server/assets/61499462/3f7876d0-54e1-4bc4-9768-12f9da3ee9f5)

**Reply by tkgup** [reply]
![image](https://github.com/rabbitmq/rabbitmq-server/assets/61499462/ad213a86-f94f-4ec4-81ea-11fa8d0eb819)

cat /usr/local/lib/rabbitmq/etc/rabbitmq/rabbitmq.config
[
 {kernel,
  [
    {inet_dist_use_interface, {172,28,9,85}},
    {inet_dist_listen_min, 50000},
    {inet_dist_listen_max, 50001},
    {net_ticktime, 60}
  ]
 },
 {rabbit,
  [
    {default_pass,<<"test">>},
    {disk_free_limit,1024000000},
    {vm_memory_high_watermark,0.2767184275151077},
    {vm_memory_high_watermark_paging_ratio, 1.1},
    {heartbeat,180},
    {force_gc_timer,5000},
    {default_user,<<"test">>},
    {config_entry_decoder,[{passphrase,{file,"/temp/pass_file"}}, {cipher, aes_256_cbc}, {iterations, 10000}]},
    {cluster_partition_handling, autoheal},
    {cluster_nodes, {[rabbit@rabbitmqNode0,rabbit@rabbitmqNode1], disc}},
    {tcp_listeners, []},
    {ssl_listeners, [{"172.28.11.155", 5672}]},
    {tcp_listen_options, [binary,
                     {packet, raw},
                     {reuseaddr, true},
                     {backlog, 4096},
                     {nodelay, true},
                     {exit_on_close, false},
                     {keepalive, true}]},
    {ssl_options, [
                   {certfile,"rabbitmq-server.crt"},
                   {keyfile,"rabbitmq-server.key"},
                   {password,{encrypted,<<"test_word">>}},
                   {versions, ['tlsv1.2']},
                   {client_renegotiation, false},
                   {ciphers, [
                              {ecdh_rsa,aes_128_gcm,null,sha256}
                             ]}
                  ]}

  ]
 },
 {rabbitmq_management,
  [
    {listener,
     [
       {port, 15672},
       {ip, "127.0.0.1"}
     ]
    },
    {redirect_old_port, false},
    {rates_mode, basic}
  ]
 },
 {rabbitmq_tracing,
  [
    {directory, "/temp/rabbitmq"}
  ]
 }
].

**Reply by michaelklishin** [reply]
[syslog logging](https://www.rabbitmq.com/docs/logging#logging-to-syslog) in RabbitMQ is optional.

RabbitMQ embeds a syslog client, so it should not be binding to a port, it should be connecting to a well known (syslog) port.

According to this output, the `syslog` app is started but not used. Here is [what it does on startup](https://github.com/schlagert/syslog/blob/master/src/syslog_logger.erl#L238-L246).

The simplest workaround I can think of is to make RabbitMQ stop the `syslog` app if syslog logging
is not configured for the node. But that might potentially break certain plugins.

Another would be to start it manually and not as a RabbitMQ server dependency. This may have
ramifications for the Bazel build system, we'd need to check if that may affect the dependency graph.

Those who care about this behavior of the Erlang `syslog` client are welcome to submit a PR, option 2 likely will have a higher chance of being accepted.

**Comment by michaelklishin** [comment]
RabbitMQ 3.11 is [out of community support](https://www.rabbitmq.com/docs/versions).
